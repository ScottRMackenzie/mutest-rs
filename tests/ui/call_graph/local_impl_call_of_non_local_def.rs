//@ print-targets
//@ stdout
//@ stderr: empty

fn clone_impl() {}

struct ImplsClone;

impl Clone for ImplsClone {
    fn clone(&self) -> Self {
        clone_impl();
        Self
    }
}

#[test]
fn test() {
    let _ = ImplsClone.clone();
}
