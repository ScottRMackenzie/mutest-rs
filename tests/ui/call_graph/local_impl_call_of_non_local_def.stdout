tests -(1)-> clone_impl at tests/ui/call_graph/local_impl_call_of_non_local_def.rs:5:1: 5:16 (#0)
  (1) test

tests -(0)-> <ImplsClone as std::clone::Clone>::clone at tests/ui/call_graph/local_impl_call_of_non_local_def.rs:10:5: 10:28 (#0)
  (0) test

targets: 2 total; 2 safe; 0 unsafe (0 tainted)
